File: src/com/owlrobotics/pedropathingvisualizer/pedropathing/util/XYLayout.java
package com.owlrobotics.pedropathingvisualizer.pedropathing.util;

import java.awt.*;

public class XYLayout implements LayoutManager {

    public XYLayout() {
    }

    @Override
    public void addLayoutComponent(String name, Component comp) {

    }

    @Override
    public void removeLayoutComponent(Component comp) {

    }

    @Override
    public Dimension preferredLayoutSize(Container parent) {
        return null;
    }

    @Override
    public Dimension minimumLayoutSize(Container parent) {
        return null;
    }

    @Override
    public void layoutContainer(Container parent) {
        int nComps = parent.getComponentCount();

        for (int i = 0; i < nComps; i++) {
            Component c = parent.getComponent(i);

            c.setSize(c.getPreferredSize());
        }
    }
}

File: src/com/owlrobotics/pedropathingvisualizer/pedropathing/util/XYLayout.kt
package com.owlrobotics.pedropathingvisualizer.pedropathing.util

import java.awt.Component
import java.awt.Container
import java.awt.Dimension
import java.awt.LayoutManager

class XYLayout : LayoutManager {
    override fun addLayoutComponent(name: String, comp: Component) {
    }

    override fun removeLayoutComponent(comp: Component) {
    }

    override fun preferredLayoutSize(parent: Container): Dimension? {
        return null
    }

    override fun minimumLayoutSize(parent: Container): Dimension? {
        return null
    }

    override fun layoutContainer(parent: Container) {
        val nComps = parent.componentCount

        for (i in 0 until nComps) {
            val c = parent.getComponent(i)

            c.size = c.preferredSize
        }
    }
}
